### PR TITLE
Refactor wifi shell

### PR DESCRIPTION
**Pull request by jukkar**
* Rework the wifi-shell code so that external subcommands can be bolted into it easily. This means that the commands should be created using SHELL_SUBCMD_ADD(). This is related to work at #78048.
* Place the various subcommands in alphabetical order to find them more easily.
* Split long lines in order to pass CI checks.
* Fixed missing newlines
* While testing I noticed that the `wifi 11v_btm_query` did not work at all, fixed it.

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -1671,14 +1671,17 @@ static int cmd_wifi_btm_query(const struct shell *sh, size_t argc, char *argv[])
 {
 	struct net_if *iface = net_if_get_first_wifi();
 	uint8_t query_reason = 0;
+	long tmp = 0;
 
 	context.sh = sh;
 
-	if (!parse_number(sh, (long *)&query_reason, argv[1], NULL,
+	if (!parse_number(sh, (long *)&tmp, argv[1], NULL,
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:1678]: Since `tmp` is of type `long`, we no longer need the cast?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -2852,15 +2852,77 @@ SHELL_STATIC_SUBCMD_SET_CREATE(
 );
 #endif /* CONFIG_WIFI_NM_WPA_SUPPLICANT_DPP */
 
-SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
-	SHELL_CMD_ARG(version, NULL, "Print Wi-Fi Driver and Firmware versions\n",
-		  cmd_wifi_version,
-		  1, 0),
-	SHELL_CMD(ap, &wifi_cmd_ap, "Access Point mode commands.\n", NULL),
-	SHELL_CMD_ARG(connect, NULL,
+static int cmd_wifi_ap(const struct shell *sh, size_t argc, char *argv[])
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:2858]: Are those dummy command handlers needed? We seem to pass NULL as a handler in a few of the net shell commands, I hope that works fine?
https://github.com/zephyrproject-rtos/zephyr/blob/029540abece01361f52025534594a0b333cf0afd/subsys/net/lib/shell/cm.c#L952

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/l2/wifi/wifi_shell.c:1678]: Indeed, good point, will remove it.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 2 above)
Comment (jukkar) [reply subsys/net/l2/wifi/wifi_shell.c:2858]: Ah, that could help, will try.

Comment (jukkar) [reply subsys/net/l2/wifi/wifi_shell.c:2858]: Yep, the extra function helper are not needed, the beast works just fine without them.

**Review by rlubos — APPROVED** [review]
LGTM

**Comment by jukkar** [comment]
@krish2718 please take a look if you have time. There are no functionality changes but allows shell subcommands to be added to wifi shell from external files (enables "wifi cred" credentials shell use case).

**Review by krish2718 — APPROVED** [review]
LGTM

**Pull request merged**